plugins/omazureeventhubs/omazureeventhubs.c

```diff
@@ -1773,3 +1037,341 @@ pn_timestamp_t time_now(void)
+			}
+			// Process if found
+			if (pMsgEntry != NULL) {
+				if (pn_delivery_remote_state(pDeliveryStatus) == PN_ACCEPTED) {
+					DBGPRINTF("handleProton: PN_DELIVERY SUCCESS for MSG '%s' @%s:%s/%s\n",
+						(pnTag.start != NULL ? (char*) pnTag.start : "NULL"),
+						pData->azurehost, 
+						pData->azureport, 
+						pData->container);
+					pMsgEntry->accepted = 1;
+
+					// Increment Stats Counter
+					STATSCOUNTER_INC(ctrAzureAck, mutCtrAzureAck);
+					INST_STATSCOUNTER_INC(pData, pData->ctrAzureAck, pData->mutCtrAzureAck);
+				} else if (pn_delivery_remote_state(pDeliveryStatus) == PN_REJECTED) {
+					LogError(0, RS_RET_ERR,
+							"omazureeventhubs: PN_DELIVERY REJECTED for MSG '%s'"
+							" - @%s:%s/%s\n",
+							(pnTag.start != NULL ? (char*) pnTag.start : "NULL"),
+							pData->azurehost,
+							pData->azureport,
+							pData->container);
+					pMsgEntry->rejected = 1;
+
+					// Increment Stats Counter
+					STATSCOUNTER_INC(ctrAzureFail, mutCtrAzureFail);
+					INST_STATSCOUNTER_INC(pData, pData->ctrAzureFail, pData->mutCtrAzureFail);
+				}
+			} else {
+				DBGPRINTF("handleProton: PN_DELIVERY NO ARRAY IDX for MSG '%d'(Max '%d') - @%s:%s/%s\n",
+					pData->nMaxProtonMsgs,
+					iTagNum
+					pData->azurehost,
+					pData->azureport,
+					pData->container);
+				STATSCOUNTER_INC(ctrAzureOtherErrors, mutCtrAzureOtherErrors);
+				INST_STATSCOUNTER_INC(pData, pData->ctrAzureOtherErrors, pData->mutCtrAzureOtherErrors);
+			}
+		} else {
+			LogError(0, RS_RET_ERR,"handleProton: PN_DELIVERY HELPER ARRAY is NULL - @%s:%s/%s\n",
+					pData->azurehost,
+					pData->azureport,
+					pData->container);
+			STATSCOUNTER_INC(ctrAzureOtherErrors, mutCtrAzureOtherErrors);
+			INST_STATSCOUNTER_INC(pData, pData->ctrAzureOtherErrors, pData->mutCtrAzureOtherErrors);
+		}
+		break;
+	}
+	case PN_TRANSPORT_CLOSED:
+		DBGPRINTF("handleProton: transport closed for %s\n", pData->azurehost);
+		proton_check_condition(event, pn_transport_condition(pn_event_transport(event)), 
+			"transport closed");
+		// Disconnected
+		pData->bIsConnected = 0;
+		break;
+	case PN_CONNECTION_REMOTE_CLOSE:
+		DBGPRINTF("handleProton: connection closed for %s\n", pData->azurehost);
+		proton_check_condition(event, pn_connection_remote_condition(pn_event_connection(event)), 
+			"connection closed");
+		break;
+	case PN_SESSION_REMOTE_CLOSE:
+		DBGPRINTF("handleProton: remote session closed for %s\n", pData->azurehost);
+		proton_check_condition(event, pn_session_remote_condition(pn_event_session(event)), 
+			"remote session closed");
+		break;
+	case PN_LINK_REMOTE_CLOSE:
+	case PN_LINK_REMOTE_DETACH:
+		DBGPRINTF("handleProton: remote link closed for %s\n", pData->azurehost);
+		proton_check_condition(event, pn_link_remote_condition(pn_event_link(event)), 
+			"remote link closed");
+		break;
+	case PN_PROACTOR_INACTIVE:
+		DBGPRINTF("handleProton: INAKTIVE for %s\n", pData->azurehost);
+		break;
+#ifdef __GNU_C
```

Comment:
why the #ifdef?

Reply:
It turns of those warnings: 

`error: enumeration value ‘PN_EVENT_NONE’ not handled in switch [-Werror=switch-enum]`

I am open for other solutions

Reply:
ok, I see. Thx! I would suggest to add this info as a comment.